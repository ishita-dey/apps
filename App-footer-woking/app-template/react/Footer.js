/* eslint-disable react/jsx-filename-extension */
import React from 'react'
import { useCssHandles } from 'vtex.css-handles'
import PropTypes from 'prop-types'
import styles from './footer.css'

const CSS_HANDLES = [
  'footerContainer',
  'headerContainer',
  'header',
  'subheader',
  'links',
  'icons',
  'socialImage',
  'disclamer',
  'copyright',
  'copyrightContainer',
]

function Footer(props) {
  const handles = useCssHandles(CSS_HANDLES)
  const headers = props.title
  const { column1, column2, column3, column4 } = props
  const columns = [column1, column2, column3, column4]
  const images = props.imagesURL
  const { disclamer } = props
  const { socailLinks } = props
  const { copyright } = props
  // const storeLink = 'https://ishita--echidna.myvtex.com/'

  // console.log(socailLinks)

  const propTypes = {
    header: PropTypes.array.isRequired,
    images: PropTypes.array.isRequired,
    disclamer: PropTypes.array.isRequired,
    socailLinks: PropTypes.array.isRequired,
    copyright: PropTypes.array.isRequired,
  }

  const footerContent = headers && headers.map((x, i) => [x, columns[i]])
  const footerSocialLinks = images && images.map((y, j) => [y, socailLinks[j]])

  // console.log(footerContent)

  // console.log(link)

  return (
    <div className={styles.footerContainer}>
      {footerContent &&
        footerContent.map((data, i) => {
          return (
            <div key={i} className={handles.headerContainer}>
              <h4 className={handles.header}>{data[0]}</h4>
              <div key={i}>
                {data[1].map((final) => {
                  return (
                    <a key={i} href={final[1]} className={handles.links}>
                      <li key={i} className={handles.subheader}>
                        {final[0]}
                      </li>
                    </a>
                  )
                })}
              </div>
            </div>
          )
        })}
      <div className={handles.icons}>
        <div>
          {footerSocialLinks &&
            footerSocialLinks.map((x, i) => {
              return (
                <a href={x[1]} key={i}>
                  <img
                    src={x[0]}
                    alt="socialIcons"
                    key={i}
                    className={handles.socialImage}
                  />
                </a>
              )
            })}
        </div>
        <div className={handles.disclamer}>{disclamer}</div>
      </div>
      <div className={handles.copyrightContainer}>
        <p className={handles.copyright}>{copyright}</p>
      </div>
    </div>
  )
}

export default Footer
