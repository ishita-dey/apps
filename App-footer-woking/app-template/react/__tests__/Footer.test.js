import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure, render, mount } from 'enzyme'
import Footer from '../Footer'
import toJson from 'enzyme-to-json'

configure({ adapter: new Adapter() })

describe('Component: Footer', () => {
  const props = {
    header: ['Products', 'Owner'],
    images: [],
    disclamer: [],
    socailLinks: [],
    copyright: [],
  }
  const footerSocialLinks = { footerSocialLinks }

  test('component rendered properly', () => {
    const wrapper = render(<Footer debug />)
    expect(toJson(wrapper, { noKey: true, mode: 'deep' })).toMatchSnapshot()
  })

  it('check prop-types', () => {
    let wrapper = shallow(<Footer {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('check childAt & type', () => {
    let wrapper = shallow(<Footer />)
    // console.log(wrapper.find('.footerContainer').childAt(0).type())
    expect(wrapper.find('.footerContainer').childAt(0).type()).toBe('div')
  })

  // it('check props at correct position', () => {
  //   let wrapper = shallow(<Footer disclamer={props.disclamer} />)
  //   expect(toJson(wrapper.find('div div div')))
  // })

  // it('check props vlaue', () => {
  //   let wrapper = mount(<Footer footerSocialLinks={footerSocialLinks} />)
  //   console.log(toJson(wrapper.find('div')))
  // })

  it('check parent class exists', () => {
    let wrapper = shallow(<Footer />)
    expect(wrapper.find('.footerContainer').length).toBe(1)
  })

  it('check first child class exists out of 4', () => {
    let wrapper = shallow(<Footer />)
    expect(wrapper.find('.icons').length).toBe(1)
  })

  it('check second child class exists out of 4', () => {
    let wrapper = shallow(<Footer />)
    expect(wrapper.find('.disclamer').length).toBe(1)
  })
  it('check third child class exists out of 4', () => {
    let wrapper = shallow(<Footer />)
    expect(wrapper.find('.copyrightContainer').length).toBe(1)
  })
  it('check fourth child class exists out of 4', () => {
    let wrapper = shallow(<Footer />)
    expect(wrapper.find('.copyright').length).toBe(1)
  })
})
