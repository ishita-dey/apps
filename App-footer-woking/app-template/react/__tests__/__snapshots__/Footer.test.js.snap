// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Footer check prop-types 1`] = `
<div
  className="footerContainer"
>
  <div
    className="icons"
  >
    <div />
    <div
      className="disclamer"
    />
  </div>
  <div
    className="copyrightContainer"
  >
    <p
      className="copyright"
    />
  </div>
</div>
`;

exports[`Component: Footer component rendered properly 1`] = `
<div
  class="footerContainer"
>
  <div
    class="icons"
  >
    <div />
    <div
      class="disclamer"
    />
  </div>
  <div
    class="copyrightContainer"
  >
    <p
      class="copyright"
    />
  </div>
</div>
`;
